cmake/SetupCodeChecks.cmake

```diff
@@ -75,67 +83,93 @@ macro(blt_add_code_check_targets cfg_file)
           list(FIND _fileTypes "${_ext}" _index)
       
           if(_index GREATER -1)
-             list(APPEND _filt_sources ${_file})
+             file(RELATIVE_PATH _relpath ${CMAKE_CURRENT_BINARY_DIR} ${_file})
+             list(APPEND _filt_sources ${_relpath})
           endif()
         endforeach()
 
-        add_uncrustify_check(CFG_FILE ${cfg_file}   SRC_FILES ${_filt_sources})
-        add_uncrustify_inplace(CFG_FILE ${cfg_file} SRC_FILES ${_filt_sources})
+        blt_add_uncrustify_check(CFG_FILE ${arg_CFG_FILE}   SRC_FILES ${_filt_sources})
+        blt_add_uncrustify_inplace(CFG_FILE ${arg_CFG_FILE} SRC_FILES ${_filt_sources})
     endif()
-    
+
 endmacro(blt_add_code_check_targets)
     
 
 ##------------------------------------------------------------------------------
 ## - Macro for invoking uncrustify to check code formatting
 ##
 ## blt_add_uncrustify_check( CFG_FILE <uncrusify_configuration_file> 
+##                           FLAGS <additional_flags_to_uncrustify>
+##                           COMMENT <additional_comment_for_target_invocation>
+##                           WORKING_DIRECTORY <working_directory>
+##                           REDIRECT <redirection_commands>
 ##                           SRC_FILES <list_of_src_files_to_uncrustify> )
 ##
 ##------------------------------------------------------------------------------
 macro(blt_add_uncrustify_check)
-    
-    MESSAGE(STATUS "Creating uncrustify check target: uncrustify_check_${PROJECT_NAME}")
+        
+    message(STATUS "Creating uncrustify check target: uncrustify_check_${PROJECT_NAME}")
 
     ## parse the arguments to the macro
     set(options)
-    set(singleValueArgs CFG_FILE)
-    set(multiValueArgs SRC_FILES)
+    set(singleValueArgs CFG_FILE COMMENT WORKING_DIRECTORY)
+    set(multiValueArgs SRC_FILES FLAGS REDIRECT)
+
     cmake_parse_arguments(arg
         "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
 
+    if(${arg_WORKING_DIRECTORY})
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+
     add_custom_target("uncrustify_check_${PROJECT_NAME}"
```

Comment:
Using ``PROJECT_NAME`` means that users can only set up one uncrustify target per ``project()``. 
Are there build systems where users only use the ``project()`` command once, but would want more than one uncrustify target?

Reply:
Yea that's not a very general purpose thing to do.  Maybe default to project if one isn't given

Reply:
In the same spirit, this macro should also take a list of source files as an input, rather than implicitly using ``${PROJECT_NAME}_ALL_SOURCES}``